libraries/adaptive-expressions/src/builtinFunctions/string.ts

```diff
@@ -26,16 +27,33 @@ export class String extends ExpressionEvaluator {
      * @private
      */
     private static evaluator(): EvaluateExpressionDelegate {
-        return FunctionUtils.apply((args: any[]): string => {
-            if (typeof args[0] === 'string') {
-                return args[0];
-            }
+        return FunctionUtils.applyWithOptionsAndError(
+            (args: any[], options: Options): ValueWithError => {
+                let result: any;
+                let error: string;
+                let locale = options.locale ? options.locale : Intl.DateTimeFormat().resolvedOptions().locale;
+                if (!error) {
+                    locale = FunctionUtils.determineLocale(args, 2, locale);
+                }
+
+                if (!error) {
+                    if (typeof args[0] === 'string') {
+                        result = args[0];
+                    } else if (typeof args[0] === 'number') {
+                        result = args[0].toLocaleString(locale);
+                    } else if (args[0] instanceof Date) {
+                        result = args[0].toLocaleDateString(locale);
+                    } else {
+                        result = JSON.stringify(args[0])
+                            .replace(/(^\'*)/g, '') // remove the starting single quote
```

Comment:
These `replace` calls could be combined into `.replace(/^['"]*/g, '').replace(/['"]*$/g, '')`.

Reply:
changed